### PR TITLE
EECC suggestions

### PR DESCRIPTION
**Pull request by AlexHentschel**
suggestions for PR https://github.com/onflow/flow-go/pull/1250
* tried to consolidate the EECC logic a bit 
* added, updated, polished goDoc

### CODE REVIEW

File: state/protocol/badger/mutator.go [hunk 1]
```diff
@@ -694,50 +685,10 @@ func (m *FollowerState) epochStatus(block *flow.Header) (*flow.EpochStatus, erro
...
-			status, err := flow.NewEpochStatus(
-				parentStatus.PreviousEpoch.SetupID, parentStatus.PreviousEpoch.CommitID,
-				parentStatus.CurrentEpoch.SetupID, parentStatus.CurrentEpoch.CommitID,
-				parentStatus.NextEpoch.SetupID, parentStatus.NextEpoch.CommitID,
-			)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create epoch status for fallback epoch: %w", err)
-			}
-			return status, fmt.Errorf("entering not set up epoch: %w", errEmergencyEpochChainContinuation)
+			return nil, fmt.Errorf("missing setup event for starting next epoch: %w", errIncompleteEpochConfiguration)
```
Comment (AlexHentschel) [inline state/protocol/badger/mutator.go:688]: consolidated all EECC logic in one place in the `handleServiceEvents` method:  https://github.com/onflow/flow-go/blob/76127e13b9a05ba62c84b7f3b349f5f131998a04/state/protocol/badger/mutator.go#L750-L766

File: state/protocol/badger/mutator.go [hunk 2]
```diff
@@ -827,14 +787,6 @@ func (m *FollowerState) handleServiceEvents(block *flow.Block) ([]func(*transact
...
 	// chain finalization should halt.
 	for _, seal := range parent.Payload.Seals {
-
-		// TMP: EMERGENCY EPOCH CHAIN CONTINUATION
-		//
-		// If an epoch failure is detected, ignore all future service events.
-		if emergencyEpochChainContinuationEnabled {
-			break
-		}
-
```
Comment (AlexHentschel) [inline state/protocol/badger/mutator.go:837]: now covered by block above: https://github.com/onflow/flow-go/blob/76127e13b9a05ba62c84b7f3b349f5f131998a04/state/protocol/badger/mutator.go#L750-L766

File: state/protocol/badger/mutator.go [hunk 3]
```diff
@@ -811,9 +774,6 @@ func (m *FollowerState) handleServiceEvents(block *flow.Block) ([]func(*transact
 	}
 	counter := activeSetup.Counter
 
-	// keep track of DB operations to apply when inserting this block
-	var ops []func(*transaction.Tx) error
-
```
Comment (AlexHentschel) [inline state/protocol/badger/mutator.go:816]: moved to beginning of method `handleServiceEvents`

**Review by jordanschalm — APPROVED** [review]
Thanks Alex

File: state/protocol/badger/mutator.go [hunk 4]
```diff
@@ -783,24 +734,36 @@ func (m *FollowerState) epochStatus(block *flow.Header) (*flow.EpochStatus, erro
...
+		// the same epoch status as its parent, resulting in it being considered
+		// by the protocol state to fall in the same epoch as its parent.
+		//
+		// CAUTION: this is inconsistent with the FinalView value specified in the epoch.
 		fmt.Println("handleServiceEvents: emergency epoch chain continuation triggered", err.Error())
+		parentStatus, err := m.epoch.statuses.ByBlockID(block.Header.ParentID)
+		if err != nil {
+			return nil, fmt.Errorf("internal error constructing EECC from parent's epoch status: %w", err)
+		}
+		ops = append(ops, m.epoch.statuses.StoreTx(block.ID(), parentStatus))
```
[the remark this answers is not in the document]
Comment (jordanschalm) [reply state/protocol/badger/mutator.go]: https://github.com/onflow/flow-go/pull/1255/commits/3ffec1b634fbd697001c1b893326fcd465b8a842

**Pull request merged**